src/main/java/com/schedule/workout/workoutScheduler/services/WorkoutsService.java

```diff
@@ -22,19 +23,30 @@
     private IWorkoutsRepository workoutsRepository;
     @Autowired
     private IUsersRepository usersRepository;
+    @Autowired
+    IUserRoleRepository userRoleRepository;
+    @Autowired
+    IRolesRepository rolesRepository;
 
     //duration
     List<Integer> durationOfWorkout = asList(20, 30, 45, 60);
 
     //create  workout
     public CreateWorkoutModel createWorkout(CreateWorkoutModel createWorkoutModel) {
+    List<UserRoleDB> userRoles = userRoleRepository.findRolesByUserId(createWorkoutModel.getTrainerId());
```

Comment:
Try to extract this logic into private method so it can be reused in both create and update logic